src/stats/_statswidget.py

```diff
@@ -187,25 +188,29 @@ def _injectCSS(self):
         if util.THEME.themeurl("ladder/style.css"):
             injectWebviewCSS(self.webview.page(), util.THEME.readstylesheet("ladder/style.css"))
 
+    def set_player(self, player):
+        if self.selected_player != player:
+            self.selected_player = player
+            self.selected_loaded = False
+
     @QtCore.pyqtSlot()
     def busy_entered(self):
         # Don't display things when we're not logged in
         # FIXME - one day we'll have more obvious points of entry
         if self.client.state != client.ClientState.LOGGED_IN:
             return
 
-        me = self.client.players[self.client.login]
-        if me.league is not None:
-            self.leagues.setCurrentIndex(me.league - 1)
+        if self.selected_player is None:
+            self.selected_player = self.client.players[self.client.login]
+        if self.selected_player.league is not None:
+            self.leagues.setCurrentIndex(self.selected_player.league - 1)
         else:
             self.leagues.setCurrentIndex(5)  # -> 5 = direct to Ladder Ratings
 
-        if self.loaded:
+        if self.selected_loaded:
```

Comment:
selected_loaded is less clear IMO, loaded or should_reload would be much better.

Reply:
should_reload would be the invers logic, maybe: selected_player_loaded ?

Reply:
selected_player_changed would be one way to call it. It really depends on how we'd extend it if there were more things to update. I'd do it by having each one set this variable to false, then check only this variable to see if we need to reload (hence should_reload makes sense). Your approach is to have each updated thing set its own variable, then check and reset all of them when reloading, so selected_player_changed fits better. Pick whichever you think is best, both are pretty good.

Reply:
selected_player_changed is again invers logic,
 which had to be set to True in '__init__' which feels wrong to me.